### DISCUSSION TITLE
Is there a way to get the header *labels* ?

### DISCUSSION THREAD

**Discussion by changyeamoon** [TanStack/table#4820] [Q&A]
I see that we can grab the ids, but is there a way to grab the labels? We were able to do this in v7, but not seeing anything in the typing that would indicate we can do this in v8

heres how to grab the ids
`table
      .getHeaderGroups()
      .map((headerGroup) => headerGroup.headers)
      .flat().find((header) => header.id === 'headerId')?.id`

**Comment by talatkuyuk** [comment]
You can add the **label** into column `meta`.

```js
declare module "@tanstack/table-core" {
  interface ColumnMeta<TData extends RowData, TValue> {
    label?: string;
  }
}
```

Then, you can define a label for each column.
```js
columnHelper.accessor("mykey", {
  cell: // your implementation,
  header: // your implementation,
  footer: // your implementation,
  meta: {
    label: "My Key Label",
  },
}),
```

Then, you can get it through the `column.columnDef.meta?.label`.

Hope it is useful.
